server/sonar-server/src/main/java/org/sonar/server/component/db/ComponentDao.java

```diff
@@ -95,12 +95,19 @@ public boolean existsById(Long id, DbSession session) {
     return mapper(session).findSubProjectsByComponentUuids(keys);
   }
 
-  public List<ComponentDto> selectModulesTree(DbSession session, String rootComponentUuid) {
-    return mapper(session).selectModulesTree(rootComponentUuid, Scopes.PROJECT);
+  public List<ComponentDto> selectEnabledModulesTree(DbSession session, String rootComponentUuid) {
+    return mapper(session).selectModulesTree(rootComponentUuid, Scopes.PROJECT, true);
   }
 
-  public List<FilePathWithHashDto> selectModuleFilesTree(DbSession session, String rootComponentUuid) {
-    return mapper(session).selectModuleFilesTree(rootComponentUuid, Scopes.FILE);
+  /**
+   * Return all child and grand child modules (including disabled ones) from a given project uuid.
```

Comment:
selectEnabledModulesTree do not return children and grand children ? (BTW children instead of child if you mean plural)